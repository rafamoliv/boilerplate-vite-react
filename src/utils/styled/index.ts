import { css } from 'styled-components'

export const getColors = (prop1, prop2) => {
  return ({ theme }) => css`
    ${theme.colors[prop1][prop2]};
  `
}
