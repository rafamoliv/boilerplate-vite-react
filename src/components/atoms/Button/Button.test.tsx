import i18next from 'i18next'
import { expect, it, vi } from 'vitest'

import { queryByTestId, render, screen, userEvent } from '@/utils/tests'

import { Button } from '.'

const props = {
  onTooggle: vi.fn(),
  children: i18next.t('translation:foo')
}

describe('<Button />', () => {
  describe('Component Render', () => {
    it('render element', () => {
      const { container } = render(<Button {...props} />)
      const button = screen.getByRole('button')
      expect(button).toBeInTheDocument()
      expect(container.firstChild).toMatchSnapshot()
    })
  })

  describe('Events', () => {
    it('Click and show element', async () => {
      render(<Button {...props} />)
      const button = screen.getByRole('button')
      await userEvent.click(button)

      const content = screen.getByText(props.children)
      expect(content).toBeInTheDocument()
    })
  })

  describe('Translate', () => {
    it('Check text translation', async () => {
      const { container } = render(<Button {...props} />)
      expect(queryByTestId(container, 'button')).toBeDefined()
      expect(queryByTestId(container, 'button')).toHaveTextContent(
        props.children
      )
    })
  })
})
