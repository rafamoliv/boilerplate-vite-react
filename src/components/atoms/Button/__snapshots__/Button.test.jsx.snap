// Vitest Snapshot v1

exports[`<Button /> > Component Render > render element 1`] = `
.c0 {
  background: #3cd3c1;
  padding: 0.5rem 1rem;
  margin: 1em;
  border: 2px solid #2a9f91;
  border-radius: 3px;
  color: #000;
}

<button
  class="c0"
  data-testid="button"
>
  Welcome to React and react-i18next
</button>
`;
